src/Shape/Box.js

```diff
@@ -0,0 +1,303 @@
+
+import Path from "./Path";
+import Rect from "./Rect";
+import Whisker from "./Whisker";
+
+/**
+    @class Box
+    @extends BaseClass
+    @desc Creates SVG lines based on an array of data.
+*/
+export default class Box extends BaseClass {
+
+  /**
+      @memberof Box
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._name = "Box";
+    this._id = (d, i) => d.id !== void 0 ? d.id : i;
+    this._pathConfig = {
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._rectConfig = {
+      fill: constant("black"),
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._value = accessor("value");
+    this._whiskerMode = ["extent", "extent"];
+    this._width = constant(50);
+    this._x = constant(200);
+    this._y = constant(200);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Filters/manipulates the data array before binding each point to an SVG group.
+      @param {Array} [*data* = the data array to be filtered]
+      @private
+  */
+  _dataFilter(data) {
+    const boxes = this._box = nest().key(this._id).entries(data).map(d => {
+
+      d.data = merge(d.values);
+      const values = d.values.map(this._value);
+      values.sort((a, b) => a - b);
+      d.i = data.indexOf(d.values[0]);
+
+      d.first = quantile(values, 0.25);
+      d.median = quantile(values, 0.50);
+      d.third = quantile(values, 0.75);
+
+      const mode = this._whiskerMode;
+
+      if (mode[0] === "tukey") d.top = d.first - (d.third - d.first) * 1.5;
+      else if (mode[0] === "extent") d.top = max(values);
+      else if (typeof mode[0] === "number") d.top = min([max(values), quantile(values, mode[0] / 100)]);
+
+      if (mode[1] === "tukey") d.bottom = d.third + (d.third - d.first) * 1.5;
+      else if (mode[1] === "extent") d.bottom = min(values);
+      else if (typeof mode[1] === "number") d.bottom = max([min(values), quantile(values, mode[1] / 100)]);
+
+      d.nested = true;
+      d.__d3plusShape__ = true;
+
+      console.log("d from box: ", d);
+
+      return d;
+    });
+
+    boxes.key = d => d.key;
+    return boxes;
+
+  }
+
+  /**
+      @memberof Box
+      @desc Draws the Box.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render() {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const filteredData = this._dataFilter(this._data);
+    console.log("filteredData: ", filteredData);
+
+    function computeHeight(d) {
+      return d.third - d.first;
+    }
+
+    new Rect()
+      .data(filteredData)
+      .width(this._width)
+      .height(computeHeight)
+      .x(this._x)
+      .y(this._y)
```

Comment:
`width`, `x`, and `y` need to be calculated based on the user's data (here they would be calculated based on our computed nested data). move this calculations into `dataFilter` and store them in our constructed data object. for example, add to 59:

```js
d.x = this._x(d.data, d.i);
```

additionally, you can also move the `height` into there as well.

Reply:
You are already calculated the x and y positions as part of `filteredData`, so you should not be using the `this._x` and `this._y` accessors again. These will return errors, as the accessors may rely on the user's data, and using them like this will not pass the user data (it will pass the `filteredData` item). These lines should be removed.